winsup/utils/getprocaddr.c

```diff
@@ -0,0 +1,198 @@
+#include <stdio.h>
+#include <windows.h>
+
+/**
+ * To determine the address of kernel32!CtrlRoutine, we need to use
+ * dbghelp.dll. But we want to avoid linking statically to that library because
+ * the normal operation of cygwin-console-helper.exe (i.e. to allocate a hidden
+ * Console) does not need it.
+ *
+ * Therefore, we declare the SYMBOL_INFOW structure here, load the dbghelp
+ * library via LoadLibraryExA() and obtain the SymInitialize(), SymFromAddrW()
+ * and SymCleanup() functions via GetProcAddr().
+ * 
+ * After that a remote thread is injected into that process
+ * to send the exit code.
+ */
+
+#include <dbghelp.h>
+
+static int pid;
+static uintptr_t exit_code;
+static BOOL do_print_address;
```

Comment:
Why do we need both `do_print_address` _and_ `pid`? Since `pid` is an `int` (shouldn't it be a `DWORD`? [`GetProcessId()`](https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-getprocessid) returns a `DWORD`...), we could simply treat `pid == 0` as an indicator that we want to print the address. After all, the process with pid 0 should never be terminated, except on shutdown.

Reply:
Ok removed https://github.com/msys2/msys2-runtime/pull/31/commits/026df76554c00d6241d1f691a257755389283208.